crates/oxc_linter_plugin/Cargo.toml

```diff
@@ -0,0 +1,31 @@
+[package]
+name                   = "oxc_linter_plugin"
+version                = "0.0.0"
+publish                = false
+authors.workspace      = true
+description.workspace  = true
+edition.workspace      = true
+homepage.workspace     = true
+keywords.workspace     = true
+license.workspace      = true
+repository.workspace   = true
+rust-version.workspace = true
+categories.workspace   = true
+
+[dependencies]
+oxc_span        = { workspace = true }
+oxc_query       = { workspace = true }
+oxc_diagnostics = { workspace = true }
+oxc_semantic    = { workspace = true }
+oxc_linter      = { workspace = true }
+oxc_parser      = { workspace = true }
+oxc_allocator   = { workspace = true }
+ignore          = { workspace = true }
+
+serde     = { workspace = true, features = ["derive"] }
+trustfall = { workspace = true }
+
+mini-v8        = "0.4.1"
```

Comment:
might be controversial to add the js back into the linter but this allows us to add a lot more customization to lints, like fixes for example, or arbitrary adds and subtracts from spans, and doesn't add much overhead at all. I'd be interested to hear how this effects larger codebases.

Reply:
Interesting, I wonder if this should be introduced in another PR so I can reason about this new dependency.

Reply:
Yup, I removed all the mini-v8 code and `post_transform`'s from this pr.